Assets/Scripts/ServiceLocator.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace HarmonyQuest
+{
+    using Input;
+    using UnityEngine;
+
+    ///<summary>
+    ///The ServiceLocator is a Singleton that points scripts to whatever is managing each of the game's major subsystems.
+    ///This script should be attached to a GameObject in a scene. If a viable manager for each subsystem is attached to the same
+    ///GameObject, that manager will be used. Otherwise, the ServiceManager will do nothing, unless a manager for that subsystem is requested,
+    ///in which case, the ServiceLocator will try to start one up. Useful for testing!
+    ///</summary>
+    public class ServiceLocator : MonoBehaviour
+    {
+        public static ServiceLocator instance;
+
+        //List of managers the ServiceLocator can provide
+        IPlayerInputManager InputManager;
+        //End List of managers
+       
+        void Awake()
+        {
+            if (instance == null)
+            {
+                instance = this;

```

Comment:
See how Mitch and I handle singleton instantiation in our reviews. In order to ensure that script order of operations on awake does not matter, we check for singleton setup both in Awake and in our singleton getters (We can still manually define script orders in unity, but this is a nice failsafe). CameraController.cs has an example of this.

Reply:
Done! This is also nice in that it makes instance readonly.